src/linear_algebra/Hermitian_inner_product.lean

```diff
@@ -0,0 +1,669 @@
+          { rw two_mul,
+            have ha : abs(x ₀ y) ≥ 0,
+            { exact abs_nonneg (x ₀ y) },
+            exact le_add_of_le_of_nonneg ha ha },
+          unfold herm_norm at Hcs,
+          rw [←sub_zero (2 * sqrt ((x ₀ x).re) * sqrt ((y ₀ y).re)), le_sub] at Hcs,
+          have Hs : |x + y|*|x + y| ≤ 2 * sqrt ((x ₀ x).re) * sqrt ((y ₀ y).re) - 2 * abs (x ₀ y) + ((x ₀ x).re + (y ₀ y).re + 2 * abs (x ₀ y)),
+          { apply le_add_of_nonneg_of_le Hcs Hle },
+          rw [sub_eq_add_neg, field.add_comm (2 * sqrt ((x ₀ x).re) * sqrt ((y ₀ y).re)), 
+            field.add_assoc ((x ₀ x).re), field.add_comm, field.add_assoc,
+            field.add_assoc, add_neg_cancel_left] at Hs, 
+          have Hs' : |x + y| * |x + y| ≤ (x ₀ x).re + ((y ₀ y).re + 2 * sqrt ((x ₀ x).re) * sqrt ((y ₀ y).re)),
+          { exact Hs },
+          have He : (x ₀ x).re + ((y ₀ y).re + 2 * sqrt ((x ₀ x).re) * sqrt ((y ₀ y).re)) = (herm_norm(x) + herm_norm(y))*(herm_norm(x) + herm_norm(y)),
+          { dunfold herm_norm,
+            rw [field.right_distrib, field.left_distrib, field.left_distrib,
+              ←sqrt_mul ((inprod_self_re_nonneg x)), sqrt_mul_self ((inprod_self_re_nonneg x)),
+              ←pow_two, sqr_sqrt (inprod_self_re_nonneg y)],
+            ring },  
+          rw He at Hs',
+          apply (mul_self_le_mul_self_iff (begin intros, exact sqrt_nonneg (((x + y) ₀ (x + y)).re), end) (add_nonneg (begin intros, exact sqrt_nonneg ((x ₀ x).re), end) (begin intros, exact sqrt_nonneg ((y ₀ y).re), end))).mpr Hs'} ,    
+      
+        { suffices : |x + y| * |x + y| + 0 ≤ (x ₀ x).re + (2 * (x ₀ y).re + (y ₀ y).re) + (2 * abs (x ₀ y) - 2 * (x ₀ y).re),
+          { simpa using this} ,
+          { apply add_le_add (le_of_eq H) hle }}}
+    end}
+
+@[simp] lemma herm_norm_smul (a : ℂ) (x : α) : 
+|a • x| = abs(a)*|x| := 
+begin 
+  intros, 
+  dunfold herm_norm, 
+  rw [inprod_smul_left, inprod_smul_right, ←field.mul_assoc,
+    mul_conj, mul_re, of_real_im, zero_mul,
+    sub_zero, of_real_re, sqrt_mul (norm_sq_nonneg a)],
+  refl,
+end
+
+@[simp] lemma of_real_herm_norm_sqr (x : α) : 
+↑( |x|^2 ) = x ₀ x :=
+begin
+  dunfold herm_norm,
+  rw [sqr_sqrt (inprod_self_re_nonneg x), re_of_real (inprod_self_im x)],
+end
+
+@[simp] lemma of_real_herm_norm_mul_self (x : α) : 
+↑( |x|*|x| ) = x ₀ x :=
+begin
+  dunfold herm_norm,
+  rw [mul_self_sqrt (inprod_self_re_nonneg x), re_of_real (inprod_self_im x)],
+end
+
+noncomputable instance complex.metric_space : metric_space ℂ :=
+{ dist := λ x y, abs(x - y),
+  dist_self := by simp,
+  eq_of_dist_eq_zero := assume x y H, sub_eq_zero.mp (complex.abs_eq_zero.mp H),
+  dist_comm := begin intros, unfold dist, rw ←neg_sub, rw complex.abs_neg, end,
+  dist_triangle := abs_sub_le}
+
+noncomputable instance complex.normed_field : normed_field ℂ :=
+{ norm := abs,
+  dist_eq := by intros; refl, 
+  norm_mul := abs_mul,}
+
+noncomputable instance herm_inner_product_space.normed_space : 
```

Comment:
I think the full name of this instance is now
`herm_inner_product_space.herm_inner_product_space.normed_space`.
If you just remove the name, then Lean will figure out a name itself. Since you are in the `herm_inner_product_space` namespace, it will generate `herm_inner_product_space.normed_space`.